### PR TITLE
fix: Add some more Ruby terms

### PR DESCRIPTION
**Pull request by andyw8**
# Add/Fix Dictionary

Dictionary: ruby

## Description

Adding various Ruby terms.

## References

See commented lines.

## Checklist

- [x] By submitting this issue, you agree to follow our
      [Code of Conduct](https://github.com/streetsidesoftware/cspell-dicts/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: dictionaries/ruby/src/ruby.txt [hunk 1]
```diff
@@ -288,14 +304,16 @@ sad asdasd
...
 SecurityError
 self
 send
 session
+Shipit
+Sidekiq
 Signal
 SignalException
 simplecov
-SimpleDelegater
```
Comment (andyw8) [inline dictionaries/ruby/src/ruby.txt:298]: Removed this incorrect spelling.

File: dictionaries/ruby/src/ruby.txt [hunk 2]
```diff
@@ -257,15 +272,16 @@ rdoc
 readonly
 Record
 records
+Redcarpet
```
Comment (andyw8) [inline dictionaries/ruby/src/ruby.txt:275]: https://github.com/vmg/redcarpet

File: dictionaries/ruby/src/ruby.txt [hunk 3]
```diff
@@ -257,15 +272,16 @@ rdoc
...
 redo
 Regexp
 RegexpError
 RELEASE_DATE
 request
 Request
 rescue
-resque
```
Comment (andyw8) [inline dictionaries/ruby/src/ruby.txt:267]: Fixing the ordering.

File: dictionaries/ruby/src/ruby.txt [hunk 4]
```diff
@@ -288,14 +304,16 @@ sad asdasd
 ScanError
 scope
 ScriptError
+securerandom
 SecurityError
 self
 send
 session
+Shipit
+Sidekiq
```
Comment (andyw8) [inline dictionaries/ruby/src/ruby.txt:313]: https://github.com/mperham/sidekiq

File: dictionaries/ruby/src/ruby.txt [hunk 5]
```diff
@@ -143,6 +149,7 @@ Float
 FloatDomainError
 for
 format
+Forwardable
```
Comment (andyw8) [inline dictionaries/ruby/src/ruby.txt:152]: https://ruby-doc.org/stdlib-2.5.1/libdoc/forwardable/rdoc/Forwardable.html

File: dictionaries/ruby/src/ruby.txt [hunk 6]
```diff
@@ -209,6 +221,7 @@ NameError
 next
 nil
 NIL
+nilable
```
Comment (andyw8) [inline dictionaries/ruby/src/ruby.txt:224]: https://sorbet.org/docs/nilable-types

File: dictionaries/ruby/src/ruby.txt [hunk 7]
```diff
@@ -157,9 +164,11 @@ GetoptLong
 getpwuid
 getutc
 gsub
+Guardfile
```
Comment (andyw8) [inline dictionaries/ruby/src/ruby.txt:167]: https://github.com/guard/guard

File: dictionaries/ruby/src/ruby.txt [hunk 8]
```diff
@@ -6,6 +6,7 @@ action
 ActionController
 ActionView
 ActiveRecord
+ActiveResource
```
Comment (andyw8) [inline dictionaries/ruby/src/ruby.txt:9]: https://github.com/rails/activeresource

File: dictionaries/ruby/src/ruby.txt [hunk 9]
```diff
@@ -20,6 +21,7 @@ asset
 association
 Attachment
 attributes
+autoload
```
Comment (andyw8) [inline dictionaries/ruby/src/ruby.txt:24]: https://medium.com/rubycademy/the-autoload-method-in-ruby-11fd079562ef

File: dictionaries/ruby/src/ruby.txt [hunk 10]
```diff
@@ -33,12 +35,15 @@ break
 Callback
 camelcase
 camelize
+CanCanCan
```
Comment (andyw8) [inline dictionaries/ruby/src/ruby.txt:38]: https://github.com/CanCanCommunity/cancancan

File: dictionaries/ruby/src/ruby.txt [hunk 11]
```diff
@@ -33,12 +35,15 @@ break
...
+CarrierWave
```
Comment (andyw8) [inline dictionaries/ruby/src/ruby.txt:39]: https://github.com/carrierwaveuploader/carrierwave

File: dictionaries/ruby/src/ruby.txt [hunk 12]
```diff
@@ -33,12 +35,15 @@ break
...
 case
 Case
 casecmp
 CGI
 CGIMethods
 chdir
+chruby
```
Comment (andyw8) [inline dictionaries/ruby/src/ruby.txt:46]: https://github.com/postmodern/chruby

File: dictionaries/ruby/src/ruby.txt [hunk 13]
```diff
@@ -69,6 +74,7 @@ display
 do
 DomainName
 downcase
+downto
```
Comment (andyw8) [inline dictionaries/ruby/src/ruby.txt:77]: https://apidock.com/ruby/Integer/downto

File: dictionaries/ruby/src/ruby.txt [hunk 14]
```diff
@@ -288,14 +304,16 @@ sad asdasd
 ScanError
 scope
 ScriptError
+securerandom
```
Comment (andyw8) [inline dictionaries/ruby/src/ruby.txt:307]: https://ruby-doc.org/stdlib-2.5.1/libdoc/securerandom/rdoc/SecureRandom.html

File: dictionaries/ruby/src/ruby.txt [hunk 15]
```diff
@@ -246,8 +259,10 @@ Process
 PStore
 Queue
 quote
+Railties
```
Comment (andyw8) [inline dictionaries/ruby/src/ruby.txt:262]: https://rubygems.org/gems/railties

File: dictionaries/ruby/src/ruby.txt [hunk 16]
```diff
@@ -246,8 +259,10 @@ Process
...
 Rake
 RakeApp
+Rakefile
```
Comment (andyw8) [inline dictionaries/ruby/src/ruby.txt:265]: https://ruby.github.io/rake/doc/rakefile_rdoc.html

File: dictionaries/ruby/src/ruby.txt [hunk 17]
```diff
@@ -157,9 +164,11 @@ GetoptLong
 getpwuid
 getutc
 gsub
+Guardfile
 Hash
 hexdigest
 hidden
+httparty
```
Comment (andyw8) [inline dictionaries/ruby/src/ruby.txt:171]: https://github.com/jnunemaker/httparty

File: dictionaries/ruby/src/ruby.txt [hunk 18]
```diff
@@ -174,6 +183,8 @@ IO
 IOError
 ipaddress
 IPSocket
+Jbuilder
```
Comment (andyw8) [inline dictionaries/ruby/src/ruby.txt:186]: https://github.com/rails/jbuilder

File: dictionaries/ruby/src/ruby.txt [hunk 19]
```diff
@@ -186,6 +197,7 @@ LocalJumpError
 location
 Logger
 Mail
+MailCatcher
```
Comment (andyw8) [inline dictionaries/ruby/src/ruby.txt:200]: https://mailcatcher.me/

File: dictionaries/ruby/src/ruby.txt [hunk 20]
```diff
@@ -351,6 +369,7 @@ Transaction
 true
 TRUE
 TrueClass
+Turbolinks
```
Comment (andyw8) [inline dictionaries/ruby/src/ruby.txt:372]: https://github.com/turbolinks/turbolinks

**Comment by nschonni** [comment]
Wonder if it would make sense to split out the Gem names to a different dictionary (like the `npm`) one or just keep them separate and merge them on build. Maybe the same for the Ruby vs. Rails words

**Comment by Jason3S** [comment]
If we see a lot of changes, pulling rails out makes sense. So far there isn't much change.

**Pull request merged**